patches/server/0952-Fix-global-sound-event-gamerule-not-being-respected.patch

```diff
@@ -0,0 +1,71 @@
++                        final double soundRadiusSquared = level.spigotConfig.witherSpawnSoundRadius == 0 ? 64 * 64 : level.spigotConfig.witherSpawnSoundRadius * level.spigotConfig.witherSpawnSoundRadius; // Paper - respect global sound events gamerule
++                        if ( !this.level.getGameRules().getBoolean(GameRules.RULE_GLOBAL_SOUND_EVENTS) && distanceSquared > soundRadiusSquared ) continue; // Spigot // Paper - respect global sound events gamerule
+                         if (distanceSquared > viewDistance * viewDistance) {
+                             double deltaLength = Math.sqrt(distanceSquared);
+                             double relativeX = player.getX() + (deltaX / deltaLength) * viewDistance;
+diff --git a/src/main/java/net/minecraft/world/item/EnderEyeItem.java b/src/main/java/net/minecraft/world/item/EnderEyeItem.java
+index f4fc8ff3981555e4b560289248a9b02afc04b42f..519bad9a16eab8e343f9340f7604115ce9c09bae 100644
+--- a/src/main/java/net/minecraft/world/item/EnderEyeItem.java
++++ b/src/main/java/net/minecraft/world/item/EnderEyeItem.java
+@@ -16,6 +16,7 @@ import net.minecraft.world.entity.player.Player;
+ import net.minecraft.world.entity.projectile.EyeOfEnder;
+ import net.minecraft.world.item.context.UseOnContext;
+ import net.minecraft.world.level.ClipContext;
++import net.minecraft.world.level.GameRules;
+ import net.minecraft.world.level.Level;
+ import net.minecraft.world.level.block.Block;
+ import net.minecraft.world.level.block.Blocks;
+@@ -69,11 +70,12 @@ public class EnderEyeItem extends Item {
+                     // world.b(1038, blockposition1.c(1, 0, 1), 0);
+                     int viewDistance = world.getCraftServer().getViewDistance() * 16;
+                     BlockPos soundPos = blockposition1.offset(1, 0, 1);
+-                    for (ServerPlayer player : world.spigotConfig.endPortalSoundRadius > 0 ? ((ServerLevel) world).players() : world.getServer().getPlayerList().players) { // Paper
++                    for (ServerPlayer player : !world.getGameRules().getBoolean(GameRules.RULE_GLOBAL_SOUND_EVENTS) ? ((ServerLevel) world).players() : world.getServer().getPlayerList().players) { // Paper - respect global sound events gamerule
+                         double deltaX = soundPos.getX() - player.getX();
+                         double deltaZ = soundPos.getZ() - player.getZ();
+                         double distanceSquared = deltaX * deltaX + deltaZ * deltaZ;
+-                        if (world.spigotConfig.endPortalSoundRadius > 0 && distanceSquared > world.spigotConfig.endPortalSoundRadius * world.spigotConfig.endPortalSoundRadius) continue; // Spigot
++                        final double soundRadiusSquared = world.spigotConfig.endPortalSoundRadius == 0 ? 64 * 64 : world.spigotConfig.endPortalSoundRadius * world.spigotConfig.witherSpawnSoundRadius; // Paper - respect global sound events gamerule
```

Comment:
Should be endPortalSoundRadius * endPortalSoundRadius (actually it's endPortalSoundRadius * witherSpawnSoundRadius)